File: CPP/coolakt/1456. Maximum Number of Vowels in a Substring of Given Length.cpp
class Solution {
public:
    int maxVowels(string s, int k) {
        int count=0;
        int mexi=0;
        for(int i=0; i<k; i++){
            if(s[i]=='a' || s[i]=='e' || s[i]=='i' || s[i]=='o' || s[i]=='u'){
                count+=1;
            }
        }
        mexi=max(count,mexi);
        for(int j=k; j<s.size(); j++){
            
            if(s[j-k]=='a' || s[j-k]=='e' || s[j-k]=='i' || s[j-k]=='o' || s[j-k]=='u'){
                count-=1;
            }
            if(s[j]=='a' || s[j]=='e' || s[j]=='i' || s[j]=='o' || s[j]=='u'){
                count+=1;
            }
            mexi=max(count,mexi);
        }
        return mexi;
    }
};

File: Java/Akash-212/1456. Maximum Number of Vowels in a Substring of Given Length.java
class Solution {
    public int maxVowels(String s, int k) {
         String vowels = "aeiou";
        int max = 0;
        int count = 0;
        for (int i = 0; i < k; i++) {
            if (vowels.indexOf(s.charAt(i)) != -1) {
                count++;
            }
        }
        max = count;
        for (int i = k; i < s.length(); i++) {
            char outgoingChar = s.charAt(i - k);
            char incomingChar = s.charAt(i);
            
            if (vowels.indexOf(outgoingChar) != -1) {
                count--;
            }
            if (vowels.indexOf(incomingChar) != -1) {
                count++;
            }
            
            max = Math.max(max, count);
        }

        return max;
    }
}
